src/ReturnTypes/CollectionFilterRejectDynamicReturnTypeExtension.php

```diff
@@ -30,7 +33,7 @@ public function getClass(): string
 
     public function isMethodSupported(MethodReflection $methodReflection): bool
     {
-        return $methodReflection->getName() === 'filter';
+        return in_array($methodReflection->getName(), ['filter', 'reject']);
```

Comment:
```suggestion
        return in_array($methodReflection->getName(), ['filter', 'reject'], true);
```

Let's be more strict.